### ISSUE TITLE
Undefined index in Matrix Field

### ISSUE THREAD

**Issue by vimcat** [Pennebaker/craft-architect#12]
Thanks for the quick fix on the previous issue. I was able to export my categories. The matrix fields that contains them, are still failing:

yii\base\ErrorException: Undefined index: source in .../vendor/pennebaker/craft-architect/src/base/FieldProcessor.php:295
Stack trace:
 .../vendor/craftcms/cms/src/web/ErrorHandler.php(84): yii\base\ErrorHandler->handleError(8, 'Undefined index...', '...', 295)

1. .../vendor/pennebaker/craft-architect/src/base/FieldProcessor.php(295): craft\web\ErrorHandler->handleError(8, 'Undefined index...', '...', 295, Array)

![screen shot 2018-03-06 at 9 48 30 am](https://user-images.githubusercontent.com/76280/37048715-8eee21ba-2123-11e8-98ae-09be8e1b5d77.png)

2. .../vendor/pennebaker/craft-architect/src/base/FieldProcessor.php(418): pennebaker\architect\base\FieldProcessor->unmapSources(Array)

![screen shot 2018-03-06 at 9 47 04 am](https://user-images.githubusercontent.com/76280/37048652-5e58818a-2123-11e8-92f6-252007f2205b.png)

3. .../vendor/pennebaker/craft-architect/src/base/FieldProcessor.php(388): pennebaker\architect\base\FieldProcessor->export(Object(craft\fields\Categories), Array, true)

![screen shot 2018-03-06 at 9 46 35 am](https://user-images.githubusercontent.com/76280/37048629-4a8842ee-2123-11e8-8c61-aca7106353fd.png)

4. .../vendor/pennebaker/craft-architect/src/base/FieldProcessor.php(432): 

![screen shot 2018-03-06 at 9 45 54 am](https://user-images.githubusercontent.com/76280/37048606-342c6476-2123-11e8-8901-8cec0f2d602f.png)

5. pennebaker\architect\base\FieldProcessor->export(Object(craft\fields\Matrix))

![screen shot 2018-03-06 at 9 45 11 am](https://user-images.githubusercontent.com/76280/37048577-219386e6-2123-11e8-8622-336e071fadfe.png)

6. .../vendor/pennebaker/craft-architect/src/controllers/DefaultController.php(128): pennebaker\architect\base\FieldProcessor->exportById('24')

**Comment by vimcat** [comment]
Nevermind, I was working with two separate command lines, and forgot to update one of them. It works great, thanks! Looking forward to the command line and migrations feature.

**Comment by spAnser** [comment]
NP. The problem appeared when I nested the field settings into the typesettings key. The weird thing is that Craft expects settings to be under typesettings but only when dealing with fields nested under other fields. It worked both ways when dealing with matrixes but when I implemented super table it needed it as typesettings so I had to go back and just make anything that was a nested field put settings under typesettings.
